lib/resty/radixtree.lua

```diff
@@ -739,6 +739,11 @@ local function match_route_opts(route, opts, args)
         if host then
             local len = #hosts
             for i = 1, len, 2 do
+                if str_find(hosts[i+1], ":", 1, true) then
+                    if opts.vars.http_host then
+                        host = opts.vars.http_host
```

Comment:
1. `host` should be reset to `opts.host` before next loop. 
2. `str_find(hosts[i+1], ":", 1, true)` could be done before route inserted into the radixtree like processing 'wildcard' for better performance.